packages/e2e-tests/specs/undo.test.js

```diff
@@ -63,4 +65,37 @@ describe( 'undo', () => {
 		// After undoing every action, there should be no more undo history.
 		expect( await page.$( '.editor-history__undo[aria-disabled="true"]' ) ).not.toBeNull();
 	} );
+
+	it( 'should undo for explicit persistence editing post', async () => {
+		// Regression test: An issue had occurred where the creation of an
+		// explicit undo level would interfere with blocks values being synced
+		// correctly to the block editor.
+		//
+		// See: https://github.com/WordPress/gutenberg/issues/14950
+
+		// Issue is demonstrated from an edited post: create, save, and reload.
+		await clickBlockAppender();
+		await page.keyboard.type( 'Hello' );
+		await pressKeyWithModifier( 'primary', 's' );
+		await page.reload();
+
+		// Issue is demonstrated by forcing state merges (multiple inputs) on
+		// an existing text after a fresh reload.
+		await selectBlockByClientId( ( await getAllBlocks() )[ 0 ].clientId );
+		await page.keyboard.press( 'End' );
```

Comment:
Does it matter whether or not the input happens at the start or end?

Reply:
> Does it matter whether or not the input happens at the start or end?

I guess it doesn't matter. More on the point of the actual interaction of appending " World" to make "Hello World" 🤷‍♂️